tests/v2/validation/rbac/globalrolesv2/globalroles_v2.go

```diff
@@ -0,0 +1,196 @@
+	return crtbs, nil
+}
+
+func getCRBsForCRTBs(client *rancher.Client, crtbs *v3.ClusterRoleTemplateBindingList) (*rbacv1.ClusterRoleBindingList, error) {
+	var downstreamCRBs rbacv1.ClusterRoleBindingList
+	for _, crtb := range crtbs.Items {
+		labelKey := fmt.Sprintf("%s_%s", crtb.ClusterName, crtb.Name)
+		req, err := labels.NewRequirement(labelKey, selection.In, []string{bindingLabel})
+		if err != nil {
+			return nil, err
+		}
+
+		selector := labels.NewSelector().Add(*req)
+		downstreamCRBsForCRTB, err := rbac.ListClusterRoleBindings(client, localcluster, metav1.ListOptions{
+			LabelSelector: selector.String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+		downstreamCRBs.Items = append(downstreamCRBs.Items, downstreamCRBsForCRTB.Items...)
+	}
+	return &downstreamCRBs, nil
+}
+
+func getRBsForCRTBs(client *rancher.Client, crtbs *v3.ClusterRoleTemplateBindingList) (*rbacv1.RoleBindingList, error) {
+	var downstreamRBs rbacv1.RoleBindingList
+	for _, crtb := range crtbs.Items {
+		roleTemplateName := crtb.RoleTemplateName
+		if strings.Contains(roleTemplateName, "rt") {
+			listOpt := metav1.ListOptions{
+				FieldSelector: "metadata.name=" + roleTemplateName,
+			}
+			roleTemplateList, err := rbac.ListRoleTemplates(client, listOpt)
+			if err != nil {
+				return nil, err
+			}
+			roleTemplateName = roleTemplateList.Items[0].RoleTemplateNames[0]
+
+		}
+		nameSelector := fmt.Sprintf("metadata.name=%s-%s", crtb.Name, roleTemplateName)
+		namespaceSelector := fmt.Sprintf("metadata.namespace=%s", crtb.ClusterName)
+		combinedSelector := fmt.Sprintf("%s,%s", nameSelector, namespaceSelector)
+
+		downstreamRBsForCRTB, err := rbac.ListRoleBindings(client, localcluster, "", metav1.ListOptions{
+			FieldSelector: combinedSelector,
+		})
+		if err != nil {
+			return nil, err
+		}
+		downstreamRBs.Items = append(downstreamRBs.Items, downstreamRBsForCRTB.Items...)
+	}
+	return &downstreamRBs, nil
+}
+
+func createDownstreamCluster(client *rancher.Client, clusterType string) (*management.Cluster, *v1.SteveAPIObject, *clusters.ClusterConfig, error) {
+	provisioningConfig := new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, provisioningConfig)
+	nodeProviders := provisioningConfig.NodeProviders[0]
+	externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviders)
+
+	testClusterConfig := clusters.ConvertConfigToClusterConfig(provisioningConfig)
+	testClusterConfig.CNI = provisioningConfig.CNIs[0]
+
+	var clusterObject *management.Cluster
+	var steveObject *v1.SteveAPIObject
+	var err error
+
+	switch clusterType {
+	case "RKE1":
+		nodeAndRoles := []provisioninginput.NodePools{
+			provisioninginput.AllRolesNodePool,
+		}
+		testClusterConfig.NodePools = nodeAndRoles
+		testClusterConfig.KubernetesVersion = provisioningConfig.RKE1KubernetesVersions[0]
+		clusterObject, _, err = provisioning.CreateProvisioningRKE1CustomCluster(client, &externalNodeProvider, testClusterConfig)
+	case "RKE2":
```

Comment:
what about k3s? is it unsupported? Same for imported clusters or hosted?

Reply:
All that we are trying to validate in the Global Roles v2 tests is the ability of the user with cluster owner (inherited role) to create clusters. It doesn't matter what cluster type it is.